### PR TITLE
[ESIMD] Disable the test on Windows as required options are not supported by clang-cl

### PR DESCRIPTION
**Pull request by fineg74**

### CODE REVIEW

File: sycl/test-e2e/ESIMD/lsc/lsc_usm_store_u32.cpp [hunk 1]
```diff
@@ -6,8 +6,8 @@
 //
 //===----------------------------------------------------------------------===//
 // REQUIRES: gpu-intel-dg2
-// DEFINE: %{winBuildFlag} =  %if windows %{-Qfp-speculation=safe%} %else %{%}
-// RUN: %{build} %{winBuildFlag}-o %t.out
+// UNSUPPORTED: windows
```
Comment (sarnex) [inline sycl/test-e2e/ESIMD/lsc/lsc_usm_store_u32.cpp:9]: Can we add a comment on why it's unsupported

**Review by sarnex — APPROVED** [review]

File: sycl/test-e2e/ESIMD/lsc/lsc_usm_store_u32.cpp (hunk 1 above)
Comment (sarnex) [reply sycl/test-e2e/ESIMD/lsc/lsc_usm_store_u32.cpp:9]: thx!

File: sycl/test-e2e/ESIMD/lsc/lsc_usm_store_u32.cpp [hunk 2]
```diff
@@ -6,10 +6,17 @@
...
+// RUN: %{build} -o %t.out
 // RUN: %{run} %t.out
 
+// TODO: Windows compiler causes this test to fail due to handling of floating
+// point arithmetics. Fixing requires using -Qfp-speculation=safe option to
+// disable some floating point optimizations to produce corect resut. icx
```
Comment (aelovikov-intel) [inline sycl/test-e2e/ESIMD/lsc/lsc_usm_store_u32.cpp:15]:
```suggestion
// disable some floating point optimizations to produce corect resut. icx
```

icx?

Comment (asudarsa) [inline sycl/test-e2e/ESIMD/lsc/lsc_usm_store_u32.cpp:15]:
```suggestion
// disable some floating point optimizations to produce correct result. icx
```

**Comment by sarnex** [comment]
This code is owned by ESIMD and I approved it, merging now to unblock testing.

**Pull request merged**